pkg/actions/addon/addon.go

```diff
@@ -94,13 +95,21 @@ func (a *Manager) getLatestVersion(addon *api.Addon) (string, error) {
 		return "", fmt.Errorf("no versions available for %q", addon.Name)
 	}
 
+	addonVersion := addon.Version
 	var versions []*version.Version
 	for _, addonVersionInfo := range addonInfos.Addons[0].AddonVersions {
-		v, err := version.NewVersion(*addonVersionInfo.AddonVersion)
+		v, err := a.parseVersion(*addonVersionInfo.AddonVersion)
 		if err != nil {
-			return "", fmt.Errorf("failed to parse version %q: %w", *addonVersionInfo.AddonVersion, err)
+			return "", err
 		}
-		versions = append(versions, v)
+
+		if addonVersion == "latest" || strings.Contains(*addonVersionInfo.AddonVersion, addonVersion) {
+			versions = append(versions, v)
+		}
+	}
+
+	if len(versions) == 0 {
+		return "", fmt.Errorf("version %s not available for %q", addonVersion, addon.Name)
```

Comment:
shall we change this to something like "no versions found matching %q"

Reply:
sure, that's better wording!